customer-managed/azure/privatelink/README.md

```diff
@@ -0,0 +1,244 @@
+# Redpanda Azure private link
+
+
+### Getting Started
+
+#### Required tools:
+- rpk
+- terraform
+- Azure CLI
+- Curl
+- jq
+
+Once all tools are installed make sure you have logged into to the azure CLI
+
+```
+az login
+```
+
+## Provisioning Redpanda with Private Link
+
+To provision a redpanda with private link there is currently only API support with UX support coming soon. 
+
+### Environment setup
+
+1. To start set the following env variables:
+    ```
+    export API_ENDPOINT=https://api.cloud.redpanda.com
+    export RPK_CLOUD_AUTH_URL=https://auth.prd.cloud.redpanda.com
+    export RPK_CLOUD_AUTH_AUDIENCE=cloudv2-production.redpanda.cloud
+    ```
+
+2. Next obtaion a client id and secret by going to Organizations in the Cloud UI. Navigate to Clients. Select or create a client you would like to use to create the network and cluster. You can copy the ID and secret to your clipboard by clicking the “Copy ID” and “Copy secret” buttons.
+    ```
+    export CLOUD_CLIENT_ID=<client-ID>
+    export CLOUD_CLIENT_SECRET=<client-secret>
+    ```
+
+3. Set the Azure subscriptions you would like to use:
+
+    ```
+    # Where the redpanda cluster will be provisioned
+    AZURE_SUBSCRIPTION_ID=60fc0bed-3072-4c53-906a-d130a934d520
+
+    # Where you will connect to the private link service fronting redpanda
+    AZURE_PRIVATE_LINK_SUBSCRIPTION_ID=1b88eb19-4c80-4edd-870d-461e83ddcbb5
+    ```
+
+### Provisioning
+
+1. Obtain an auth token
+    ```
+    export AUTH_TOKEN=`curl -s --request POST \
+        --url "${RPK_CLOUD_AUTH_URL}/oauth/token" \
+        --header 'content-type: application/x-www-form-urlencoded' \
+        --data grant_type=client_credentials \
+        --data client_id=$CLOUD_CLIENT_ID \
+        --data client_secret=$CLOUD_CLIENT_SECRET \
+        --data audience=$RPK_CLOUD_AUTH_AUDIENCE | jq -r '.access_token'`
+
+    # MAKE SURE THIS IS POUPULATED
+    echo $AUTH_TOKEN
+
+
+    declare -a HEADERS=(
+        "-H" "Content-Type: application/json"
+        "-H" "Authorization: Bearer $AUTH_TOKEN"
+    )
+
+    ```
+
+1. Find the Resource Group where we will create the redpanda cluster
+
+    ```
+    RESOURCE_GROUP_ID=`curl -s -X GET "${HEADERS[@]}" \
+    $API_ENDPOINT/v1beta2/resource-groups | jq -r ".resource_groups[0].id"`
+
+    # NOTE: this should have output
+    echo $RESOURCE_GROUP_ID
+
+    ```
+
+1. Create the network that will be used by the redpanda cluster
+    ```
+    NETWORK_POST_BODY=`cat << EOF
+    {
+    "cidr_block": "10.0.0.0/20",
```

Comment:
suggestion, nonblocking: the customer may not be using a 10.0.0.0 network, or `uksouth` region